core-httpclient-impl/src/test/java/com/optimizely/ab/odp/DefaultODPApiManagerTest.java

```diff
@@ -0,0 +1,116 @@
+
+        when(mockHttpClient.execute(any(HttpPost.class)))
+            .thenReturn(httpResponse);
+    }
+
+    @Test
+    public void generateCorrectSegmentsStringWhenListHasOneItem() {
+        DefaultODPApiManager apiManager = new DefaultODPApiManager();
+        String expected = "\\\"only_segment\\\"";
+        String actual = apiManager.getSegmentsStringForRequest(Arrays.asList("only_segment"));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void generateCorrectSegmentsStringWhenListHasMultipleItems() {
+        DefaultODPApiManager apiManager = new DefaultODPApiManager();
+        String expected = "\\\"segment_1\\\", \\\"segment_2\\\", \\\"segment_3\\\"";
+        String actual = apiManager.getSegmentsStringForRequest(Arrays.asList("segment_1", "segment_2", "segment_3"));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void generateEmptyStringWhenGivenListIsEmpty() {
+        DefaultODPApiManager apiManager = new DefaultODPApiManager();
+        String actual = apiManager.getSegmentsStringForRequest(new ArrayList<>());
+        assertEquals("", actual);
+    }
+
+    @Test
+    public void generateCorrectRequestBody() throws Exception {
+        ODPApiManager apiManager = new DefaultODPApiManager(mockHttpClient);
+        apiManager.fetchQualifiedSegments("key", "endPoint", "fs_user_id", "test_user", Arrays.asList("segment_1", "segment_2"));
+        verify(mockHttpClient, times(1)).execute(any(HttpPost.class));
+
+        String expectedResponse = "{\"query\": \"query {customer(fs_user_id: \\\"test_user\\\") {audiences(subset: [\\\"segment_1\\\", \\\"segment_2\\\"]) {edges {node {name state}}}}}\"}";
+        ArgumentCaptor<HttpPost> request = ArgumentCaptor.forClass(HttpPost.class);
+        verify(mockHttpClient).execute(request.capture());
+        assertEquals(expectedResponse, EntityUtils.toString(request.getValue().getEntity()));
+    }
+
+    @Test
+    public void returnResponseStringWhenStatusIs200() throws Exception {
+        ODPApiManager apiManager = new DefaultODPApiManager(mockHttpClient);
+        String responseString = apiManager.fetchQualifiedSegments("key", "endPoint", "fs_user_id", "test_user", Arrays.asList("segment_1", "segment_2"));
+        verify(mockHttpClient, times(1)).execute(any(HttpPost.class));
+        assertEquals(validResponse, responseString);
+    }
+
+    @Test
+    public void returnNullWhenStatusIsNot200AndLogError() throws Exception {
+        setupHttpClient(500);
+        ODPApiManager apiManager = new DefaultODPApiManager(mockHttpClient);
+        String responseString = apiManager.fetchQualifiedSegments("key", "endPoint", "fs_user_id", "test_user", Arrays.asList("segment_1", "segment_2"));
+        verify(mockHttpClient, times(1)).execute(any(HttpPost.class));
+        logbackVerifier.expectMessage(Level.ERROR, "Unexpected response from ODP server, Response code: 500, null");
+        assertEquals(null, responseString);
+    }
+}
```

Comment:
What about the test for network issues, such as internet is down, where status code is not returned. Request is not reaching the server and bounces off?